src/components/AddDataset.tsx

```diff
@@ -0,0 +1,93 @@
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { useState } from 'react';
+
+interface ShowProps {
+  show: boolean;
+  toggleModal: () => void;
+}
+
+const AddDataset: React.FC<ShowProps> = (props: ShowProps) => {
+  const [name, setName] = useState('');
```

Comment:
Since we are already using JSON forms we might as well use it again here. It will handle things like validation and handling changes